lib/galaxy/tools/cache.py

```diff
@@ -161,11 +161,11 @@ def cleanup(self):
         removed_tool_ids = []
         try:
             with self._lock:
-                persist_tool_document_cache = False
+                persist_callback = None
                 paths_to_cleanup = {(path, tool) for path, tool in self._tools_by_path.items() if self._should_cleanup(path)}
                 for config_filename, tool in paths_to_cleanup:
                     tool.remove_from_cache()
-                    persist_tool_document_cache = True
+                    persist_callback = tool.app.toolbox.persist_cache
```

Comment:
I preferred the previous version using a boolean to track if we need to persist, which is also slightly more efficient being in a loop?